### PR TITLE
Initial intake

### LINKED ISSUE #19: Make Intake Subsystem
Added intake logic and commands.

### PR DESCRIPTION
**Pull request by bobtheragelot**
Added Initial Intake code
 
* RunIntake Command
-Bound to X
* ToggleJoint Command
-Bound to Y
-States shown on SmartDashboard

resolve(#19 )

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -26,15 +33,18 @@
 public class RobotContainer {
 
   private static XboxController driveController;
+  private static XboxController altController;
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/RobotContainer.java]: no static

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -57,6 +67,13 @@ public RobotContainer() {
    */
   private void configureButtonBindings() {
     driveController = new XboxController(driveControllerPort);
+    altController = new XboxController(altControllerPort);
+
+    JoystickButton toggleIntake = new JoystickButton(altController, Button.kX.value);
+    toggleIntake.toggleWhenPressed(new RunIntake(IntakeConstants.intakeMotorPower));
+
+    JoystickButton toggleJoint = new JoystickButton(altController, Button.kY.value);
+    toggleJoint.whenPressed(new ToggleJoint(IntakeConstants.jointMotorPower).withTimeout(1));
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/RobotContainer.java]: toggleJointPosition?

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -57,6 +67,13 @@ public RobotContainer() {
    */
   private void configureButtonBindings() {
     driveController = new XboxController(driveControllerPort);
+    altController = new XboxController(altControllerPort);
+
+    JoystickButton toggleIntake = new JoystickButton(altController, Button.kX.value);
+    toggleIntake.toggleWhenPressed(new RunIntake(IntakeConstants.intakeMotorPower));
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/RobotContainer.java]: change to defaultMotorPower.
Also import static Constants.IntakeConstants so you dont have to do IntakeConstants. every time

File: src/main/java/frc/robot/commands/RunIntake.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Intake;
+
+
+public class RunIntake extends CommandBase {
+
+  Intake intake;
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/commands/RunIntake.java]: should be private

File: src/main/java/frc/robot/commands/RunIntake.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  public RunIntake(double motorPower) {
+    intake = Intake.getInstance();
+    addRequirements(intake);
+
+    this.motorPower = motorPower;
+  }
+
+  @Override
+  public void execute() {
+    intake.runIntake(this.motorPower);
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/commands/RunIntake.java]: this. not needed

File: src/main/java/frc/robot/commands/RunIntake.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    return false;
+  }
+
+  @Override
+  public void end(boolean interrupted) {
+    intake.runIntake(0);
+  }
+}
+
+
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/commands/RunIntake.java:36]: heck this whitespace

File: src/main/java/frc/robot/commands/ToggleJoint.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.Intake.PivotState;
+
+public class ToggleJoint extends CommandBase {
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/commands/ToggleJoint.java:7]: ToggleJointPosition or State or something idk mane

File: src/main/java/frc/robot/commands/ToggleJoint.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+  private Intake intake;
+  private double motorPower;
+
+  public ToggleJoint(double power) {
+    motorPower = power;
+
+    intake = Intake.getInstance();
+
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/commands/ToggleJoint.java:16]: make less wack spacing

File: src/main/java/frc/robot/commands/ToggleJoint.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  public void execute() {
+    if (intake.getPivotState() == PivotState.down) {
+      intake.runJoint(-motorPower);
+    } else {
+      intake.runJoint(motorPower);
+    }
+  }
+
+  /**
+   * If PivotState is down, return state of limit switch, which should be false.
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/commands/ToggleJoint.java]: Bad comment.  You are describing the code in words, describe why it happens not how it happens

File: src/main/java/frc/robot/subsystems/DriveTrain.java [hunk 10]
```diff
@@ -37,6 +37,7 @@
   private WPI_VictorSPX leftFollower2;
 
   public DriveTrain() {
+
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/subsystems/DriveTrain.java]: heck this line

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 11]
```diff
@@ -7,18 +7,68 @@
...
-  /**
-   * Creates a new Intake.
-   */
+
+  private static Intake instance;
+  private static VictorSP jointMotor;
+  private static VictorSP intakeMotor;
+  private DigitalInput limitSwitch;
+  private PivotState pivotState;
+  private int num;
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/subsystems/Intake.java]: heck this num

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 12]
```diff
@@ -7,18 +7,68 @@
...
+
+  public enum PivotState {
+    up,
+    down
+  }
+
   public Intake() {
+    limitSwitch = new DigitalInput(SensorConstants.limitSwitchPort);
+    jointMotor = new VictorSP(IntakeConstants.jointMotorPort);
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/subsystems/Intake.java]: import static constants.intakeconstants

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 13]
```diff
@@ -7,18 +7,68 @@
...
+  public boolean getLimitSwitch() {
+    return limitSwitch.get();
+  }
+
+  public PivotState getPivotState() {
+    return pivotState;
+  }
+
+  public void setPivotStateDown() {
+    pivotState = PivotState.down;
```
Comment (Dustin-Klein) [inline src/main/java/frc/robot/subsystems/Intake.java]: rewrite to setPivotState(PivotState pivotState) instead of two separate setter methods

**Review by Dustin-Klein — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Intake.java (hunk 11 above)
Comment (StaleyRobotics) [reply src/main/java/frc/robot/subsystems/Intake.java]: heck u djstinm

**Review by Dustin-Klein — DISMISSED** [review]

**Review by Dustin-Klein — APPROVED** [review]

**Pull request merged**